packages/ui-components/src/components/Map/PopupDataItemList.js

```diff
@@ -28,31 +28,23 @@ PopupDataItem.propTypes = {
   measureName: PropTypes.string.isRequired,
 };
 
-export const PopupDataItemList = ({ serieses, data, showNoDataLabel }) => {
-  const popupList = data
-    ? serieses
-        .filter(series => !series.hideFromPopup)
-        .sort((measure1, measure2) => measure1.sortOrder - measure2.sortOrder)
-        .map(series => {
-          const { key, name, organisationUnit, ...otherConfigs } = series;
-          const metadata = getMetadata(data, key);
-          const { formattedValue, valueInfo } = getFormattedInfo(data, series, {
-            key,
-            metadata,
-            ...otherConfigs,
-          });
-          return valueInfo.hideFromPopup ? null : (
-            <PopupDataItem key={name} measureName={name} value={formattedValue} />
-          );
-        })
-        .filter(popupItem => popupItem !== null)
-    : [];
-
-  const { name, key } = serieses[0];
-
-  return popupList.length === 0 && showNoDataLabel
-    ? [<PopupDataItem key={name || key} measureName={name || key} value="No Data" />]
-    : popupList;
+export const PopupDataItemList = ({ serieses, data }) => {
+  return serieses
+    .filter(series => !series.hideFromPopup)
+    .sort((measure1, measure2) => measure1.sortOrder - measure2.sortOrder)
+    .map(series => {
+      const { key, name, organisationUnit, ...otherConfigs } = series;
+      const metadata = getMetadata(data, key);
+      const { formattedValue, valueInfo } = getFormattedInfo(data, series, {
+        key,
+        metadata,
+        ...otherConfigs,
+      });
+      return valueInfo.hideFromPopup ? null : (
+        <PopupDataItem key={name} measureName={name} value={formattedValue} />
+      );
+    })
+    .filter(popupItem => popupItem !== null);
```

Comment:
Mainly removing `showNoDataLabel` here, we hand back the logic to web-config-server ([link](https://github.com/beyondessential/tupaia/blob/404b0c49b2fdbd656bc55623934fc25c93f5db4f/packages/web-config-server/src/apiV1/measureData.js#L239)) to tell us when show no data label on tooltips.